src/main/kotlin/net/maizegenetics/phgv2/cli/CreateRanges.kt

```diff
@@ -49,20 +50,23 @@ class CreateRanges: CliktCommand(help="Create BED file of reference ranges from
                 genes.map { gene ->
                     gene.children
                         .filter { transcript ->
-                            transcript.attributes.containsKey("canonical_transcript")
+                            transcript.allAttributes().containsKey("canonical_transcript")
                         }
                         .flatMap { it.children }
-                        .filter { it.type().name == "CODING_SEQUENCE" }
+                        .filter { it.type == "CDS" }
                         .map { Pair(it.start, it.end) }
                         .let { ranges ->
-                            Pair(ranges.minOf { it.first }, ranges.maxOf { it.second })
+                            if(ranges.isNotEmpty())
+                                Pair(ranges.minOf { it.first }, ranges.maxOf { it.second })
+                            else
+                                Pair(0,0)
                         }
-                }
+                }.filter { it.first != 0 && it.second != 0 }
             }
             else -> throw Exception("Undefined boundary")
         }.map { (first, second) ->
             val modifiedFirst = if (first - pad < 0) 1 else first - pad
-            Pair((modifiedFirst - 1), (second + pad) - 1)
+            Pair((modifiedFirst - 1), (second + pad) )
```

Comment:
Add a comment here about 0-based vs 1-based, what was passed in and why it needs to be modified.  This will keep it clear for future people reading the code, and verifies the conversion was thought out.

Reply:
Done